torch/_tensor_str.py

```diff
@@ -277,31 +277,39 @@ def _vector_str(self):
         ident = ' '
     if self.numel() < PRINT_OPTS.threshold:
         return (strt +
-                '\n'.join(ident + fmt.format(val / scale) for val in self) +
+                '\n'.join(ident + fmt.format(val.item() / scale) for val in self) +
                 '\n')
     else:
         return (strt +
-                '\n'.join(ident + fmt.format(val / scale) for val in self[:n]) +
+                '\n'.join(ident + fmt.format(val.item() / scale) for val in self[:n]) +
                 '\n' + (ident + dotfmt.format(u"\u22EE")) +
-                '\n'.join(ident + fmt.format(val / scale) for val in self[-n:]) +
+                '\n'.join(ident + fmt.format(val.item() / scale) for val in self[-n:]) +
                 '\n')
 
 
 def _str(self, include_footer=True):
-    if self.ndimension() == 0:
+    empty = self.numel() == 0
+    dim = self.dim()
+
+    if empty:
         strt = ''
-    elif self.ndimension() == 1:
+    elif dim == 0:
+        strt = _vector_str(self.unsqueeze(0))
+    elif dim == 1:
         strt = _vector_str(self)
-    elif self.ndimension() == 2:
+    elif dim == 2:
         strt = _matrix_str(self)
     else:
         strt = _tensor_str(self)
 
     if include_footer:
-        size_str = 'x'.join(str(size) for size in self.size())
-        size_str_prefix = 'of size ' if self.ndimension() > 0 else 'with no dimension'
+        if empty:
+            size_str = '(0)'
```

Comment:
Why don't we just use what's in variable.py?
size_str = '(' + ','.join(str(size) for size in self.size()) + (',)' if len(self.size()) == 1 else ')')

This looks like sometimes it's using "x" deliminator, sometimes (), that just seems worse.

Reply:
I got rid of the `x` delimiter